crates/lsp/src/text_edit.rs

```diff
@@ -79,12 +79,6 @@ impl TextEdit {
         builder.finish()
     }
 
-    // --- Start Posit
-    pub fn diff(text: &str, replace_with: &str) -> TextEdit {
-        super::diff::diff(text, replace_with)
-    }
```

Comment:
I've moved `rust_analyzer/text_edit.rs` to just `text_edit.rs`, and `rust_analyzer/diff.rs` to just `diff.rs`, but I've left all the rust-analyzer attribution in place

I did remove this one helper in favor of just calling `crate::diff::diff()` directly, so we can keep this file as a pure rust-analyzer copy/paste